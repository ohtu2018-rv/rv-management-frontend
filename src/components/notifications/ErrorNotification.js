import React from 'react';
import PropTypes from 'prop-types';

import './styles/ErrorNotification.css';

/**
 * Error notification component.
 */
const ErrorNotification = ({ message, shadow }) => {
    return (
        <div
            className={
                shadow ? 'error-message error-message-shadow' : 'error-message'
            }
        >
            <svg className="cross" width="70" height="70">
                <path d="m35,35l-18.3,-18.3" />
                <path d="m35,35l18.3,18.3" />
                <path d="m35,35l-18.3,18.3" />
                <path d="m35,35l18.3,-18.3" />
            </svg>
            <div className="inner">
                <div className="notification-text">{message}</div>
            </div>
        </div>
    );
};

ErrorNotification.propTypes = {
    /** Notification message. */
    message: PropTypes.string.isRequired,
    /** Shadow effect under the notification */
    shadow: PropTypes.bool
};

export default ErrorNotification;
