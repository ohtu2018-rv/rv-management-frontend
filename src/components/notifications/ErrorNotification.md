### With shadow

```js
<div style={{width: 250}}><ErrorNotification message={"This is an error message"} shadow /></div>
```

### Without shadow

```js
<div style={{width: 250}}><ErrorNotification message={"This is an error message"} /></div>
```
