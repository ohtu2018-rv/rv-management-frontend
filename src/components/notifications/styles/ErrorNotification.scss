@import "vars";

.error-message {
  font-size: $notification-font-size;
  padding: $base-padding;
  text-align: center;
  display: block;
  white-space: nowrap;
  background-color: $error-notification-bgcolor;
  border-radius: $notification-border-radius;
  color: $error-notification-textcolor;
  margin: $base-margin;
}

.error-message-shadow {
  box-shadow: $box-shadow;
}

.cross {
  stroke: $error-notification-textcolor;
  stroke-linecap: round;
  stroke-width: $error-notification-cross-width;
}
