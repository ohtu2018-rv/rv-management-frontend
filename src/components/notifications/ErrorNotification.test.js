import React from 'react';
import { shallow } from 'enzyme';
import ErrorNotification from './ErrorNotification';
import renderer from 'react-test-renderer';

describe.only('<ErrorNotification />', () => {
    it('renders correctly', () => {
        const cmpnt = renderer
            .create(<ErrorNotification message="Test notification" />)
            .toJSON();
        expect(cmpnt).toMatchSnapshot();
    });
});
