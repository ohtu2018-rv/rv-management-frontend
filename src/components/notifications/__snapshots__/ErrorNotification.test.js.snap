// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorNotification /> renders correctly 1`] = `
<div
  className="error-message"
>
  <svg
    className="cross"
    height="70"
    width="70"
  >
    <path
      d="m35,35l-18.3,-18.3"
    />
    <path
      d="m35,35l18.3,18.3"
    />
    <path
      d="m35,35l-18.3,18.3"
    />
    <path
      d="m35,35l18.3,-18.3"
    />
  </svg>
  <div
    className="inner"
  >
    <div
      className="notification-text"
    >
      Test notification
    </div>
  </div>
</div>
`;
